lib/auth/password.go

```diff
@@ -190,59 +196,58 @@ func (s *Server) CheckPasswordWOToken(user string, password []byte) error {
 }
 
 // CheckPassword checks the password and OTP token. Called by tsh or lib/web/*.
-func (s *Server) CheckPassword(user string, password []byte, otpToken string) error {
+func (s *Server) CheckPassword(user string, password []byte, otpToken string) (*types.MFADevice, error) {
```

Comment:
Hmm, this makes the function signature a little strange IMO. I.e. it seems a little "random" that CheckPassword returns MFA device, though I understand why it's needed.

As a small improvement suggestion, what if we made this function return `CheckPasswordResult` struct that MFADevice would be one of the fields of (and the only field, for now at least)? Then I think the intent will be a bit clearer, and you wouldn't also need to return `nil, nil` from this function (as it is now).

Reply:
Done